### DISCUSSION TITLE
Digitally sign copilot-agent-win.exe

### DISCUSSION THREAD

**Discussion by jcoleson** [community/community#36665] [Copilot Conversations] [answered]
Can we get copilot-agent-win.exe digitally signed? 

Our corporate environment has Windows Applocker set to block unsigned EXE files from running in Windows. Copilot-agent-win.exe doesn't appear to be digitally signed and therefore is being blocked from running within Visual Studio. That means every time there is an update I have to manually sign it with my own certificate to ensure it runs properly.

**Comment by iamcjwilliams** [comment]
@jcoleson Support for self-signed certificates is a known limitation in our current proxy support. We understand that there are a number of Copilot users that work in environments where this is a necessary requirement, that's why we have support for self-signed certificates on our radar and aim to implement the functionality in the future. We are unable to share a specific timeline for adding self-signed certificate support at this point in time.

**Comment by ossanna16** [comment]
Hey there @jcoleson and thank you for asking a good question! I'm glad you found our community 🙂

**Comment by jcoleson** [comment]
I've gotten around this by digitally signing `COPILOT-AGENT-WIN.EXE` each time there was a new update. While annoying, it was a reasonable workaround. However, I now have to Sign In again. I cannot Sing-In because `copilot-find-certificates.exe` isn't signed. Because `copilot-find-certificates.exe` is running from a temp directory with a new folder name each time, I can't digitally sign it myself. 

Is there is work around to sign in to the Visual Studio Plugin? 

I'll have to cancel my account if I can't sign in anymore.

**Reply by MikkelLundKnudsen** [reply]
This is barely a solution imho, GitHub - please sign your .exe :)

**Comment by iamcjwilliams** [comment]
@jcoleson Try [here](https://docs.github.com/en/copilot/troubleshooting-github-copilot) to troubleshoot your issue

**Comment by thisisevilevil** [comment]
Bump. This is still not signed, we are rolling this out in our org, and now we get weekly requests from our devs for getting the file hash whitelisted in our AppLocker policies

**Reply by jcoleson** [reply]
Now that GitHub Copilot for Business has rolled out, I’m surprised this hasn’t been added. GitHub Copilot for Business is clearly targeted at larger organizations yet lacks support for this common security feature required in enterprise environments. 

If you need a sample business case, we had a few individuals buy their Copilot subscriptions on their own. When GitHub Copilot for Business rolled out, we decided to enroll a few of those developers at the higher price point to see if it was an improved experience. We were disappointed to see even GitHub Copilot for Business wasn’t signed. 

We‘re still deciding if we want to roll GitHub Copilot for Business out to our whole department. At then moment we’re leaning towards canceling our GitHub Copilot for Business subscriptions and going back to reimbursing individuals if they want to deal with the hassle of manually signing every single update in order for it to be usable. It says a lot about the product that several people are willing to go through with hassle. But as a team lead I can’t bring myself to recommend we pay for a business tool that works so poorly in our environment.

**Reply by MikkelLundKnudsen** [reply]
> Now that GitHub Copilot for Business has rolled out, I’m surprised this hasn’t been added. GitHub Copilot for Business is clearly targeted at larger organizations yet lacks support for this common security feature required in enterprise environments.
> 
> If you need a sample business case, we had a few individuals buy their Copilot subscriptions on their own. When GitHub Copilot for Business rolled out, we decided to enroll a few of those developers at the higher price point to see if it was an improved experience. We were disappointed to see even GitHub Copilot for Business wasn’t signed.
> 
> We‘re still deciding if we want to roll GitHub Copilot for Business out to our whole department. At then moment we’re leaning towards canceling our GitHub Copilot for Business subscriptions and going back to reimbursing individuals if they want to deal with the hassle of manually signing every single update in order for it to be usable. It says a lot about the product that several people are willing to go through with hassle. But as a team lead I can’t bring myself to recommend we pay for a business tool that works so poorly in our environment.

Totally agree, GitHub needs to sign this .exe files - asap :)

**Comment by alwell-kevin** [comment, marked as answer]
This should be fixed. Try upgrading your Copilot extension to latest. If that doesn't work, try updating your IDE to latest. 

If no luck, re-prompt this thread 👍

**Reply by jcoleson** [reply]
Thank you for letting me know!
